Configuration/NativeMethods.cs

```diff
@@ -64,7 +64,9 @@ private struct AssemblyInfo
         public static string GetAssemblyPath(string name)
         {
             if (name == null)
-                throw new ArgumentNullException("name");
+                throw new ArgumentNullException("name");
+              
+            if( Type.GetType( "Mono.Runtime" ) != null ) return null;
 
```

Comment:
I thought that GetAssemblyPath might be called from other places and thus decided to cut the problem at it source. If the only usage of that method is guaranteed to be only in AzureApplicationSettings then skipping the whole class will work.

If I recall correctly, the main problem comes when creating an instance of Settings. Because of this line:
private static readonly AzureApplicationSettings azureSettings = new AzureApplicationSettings();

It always goes to the path that uses fusion.dll. 
Maybe it would be better to not initialize the value with its declaration, but do it in the constructor with a check there (And leave it null if it does not apply)